README.md
# ArmA3_scarCODE_HUD
by IT07 <br />
######DESCRIPTION
This "script" "disables" the default vanilla ArmA 3 HUD and shows a custom HUD instead.<br />
It is very clean, light and simple.
######HOW TO DOWNLOAD
Click the "Download ZIP" button<br />
######INSTALLATION GUIDE
```
MissionName.MapName\init.sqf (contents) >> MPmissions\*missionName*.*nameOfMap*\init.sqf
MissionName.MapName\description.ext (contents) >> MPmissions\*missionName*.*nameOfMap*\description.ext
MissionName.MapName\scarCODE >> MPmissions\*missionName*.*nameOfMap*\
```


CHANGELOG.md
###April 24th
First release.


MissionName.MapName/scarCODE/scarHUD/cfg/CfgStanceImages.hpp
class CfgStanceImages
{
   class aadjpercmstpsraswrfldup // Higher stance (stand)
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_stand_up_ca.paa";
   };
   class amovpercmstpsraswrfldnon
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_stand_ca.paa";
   };
   class aadjpercmstpsraswrflddown
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_stand_down_ca.paa";
   };
   class aadjpercmstpsraswrfldleft
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_stand_left_ca.paa";
   };
   class aadjpercmstpsraswrfldright
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_stand_right_ca.paa";
   };
   class aadjpknlmstpsraswrfldup
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_crouch_up_ca.paa";
   };
   class amovpknlmstpsraswrfldnon
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_crouch_ca.paa";
   };
   class aadjpknlmstpsraswrfldleft
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_crouch_left_ca.paa";
   };
   class aadjpknlmstpsraswrfldright
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_crouch_right_ca.paa";
   };
   class aadjpknlmstpsraswrflddown
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_crouch_down_ca.paa";
   };
   class aadjppnemstpsraswrfldup
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_prone_up_ca.paa";
   };
   class amovppnemstpsraswrfldnon
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_prone_ca.paa";
   };
   class aadjppnemstpsraswrflddown
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_prone_down_ca.paa";
   };
   class aadjppnemstpsraswrfldleft
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_prone_left_ca.paa";
   };
   class aadjppnemstpsraswrfldright
   {
      image = "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\si_prone_right_ca.paa";
   };
};


MissionName.MapName/scarCODE/scarHUD/hpp/RscScarHudCurrentMuzzle.hpp
class RscScarHudCurrentMuzzle
{
    idd = 16030;
    duration = 99999;
    fadeIn = 0.3;
    onLoad = "uiNamespace setVariable ['RscScarHudCurrentMuzzle', _this select 0]";
    class IGUIBack
    {
    	type = 0;
    	style = 128;
    	text = "";
    	colorText[] = {0,0,0,0};
    	font = "PuristaMedium";
    	sizeEx = 0;
    	shadow = 0;
    	colorbackground[] = {1,1,1,0.97};
    };
    class RscText
    {
    	deletable = 0;
    	fade = 0;
    	access = 0;
    	type = 0;
    	colorBackground[] = {1,1,1,0.97};
    	colorText[] = {0,0,0,0.8};
    	text = "";
    	fixedWidth = 0;
    	style = 0;
    	shadow = 0;
    	colorShadow[] = {0,0,0,0.5};
    	font = "PuristaMedium";
    	SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1)";
    	linespacing = 1;
    	tooltipColorText[] = {1,1,1,1};
    	tooltipColorBox[] = {1,1,1,1};
    	tooltipColorShade[] = {0,0,0,0.65};
    };
    class Controls
    {
      class weaponName: RscText
      {
        idc = 999;
        x = 0.856 * safezoneW + safezoneX;
        y = 0.956 * safezoneH + safezoneY;
        w = 0.1315 * safezoneW;
        h = 0.02 * safezoneH;

        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 0.7)";
        style = 1;
      };
      class bulletCountBackground: IGUIBack
      {
        idc = -1;
        x = 0.889 * safezoneW + safezoneX;
        y = 0.917 * safezoneH + safezoneY;
        w = 0.098 * safezoneW;
        h = 0.036 * safezoneH;
      };
      class bulletCount: RscText
      {
        idc = 1001;
        style = 1;
        font = "PuristaSemiBold";
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 2.7)";
        colorBackground[] = {-1,-1,-1,-1};
        x = 0.889 * safezoneW + safezoneX;
        y = 0.911 * safezoneH + safezoneY;
        w = 0.098 * safezoneW;
        h = 0.04 * safezoneH;
      };
      class currentZeroing: RscText
      {
        idc = 1003;
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 0.8)";
        x = 0.856 * safezoneW + safezoneX;
        y = 0.915 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.018 * safezoneH;
      };
      class currentWeaponMode: RscText
      {
        idc = 1004;
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 0.8)";
        x = 0.856 * safezoneW + safezoneX;
        y = 0.935 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.018 * safezoneH;
      };
      class magazineCount: RscText
      {
        idc = 1005;
        style = 1;
        font = "PuristaBold";
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1.6)";
        x = 0.889 * safezoneW + safezoneX;
        y = 0.892 * safezoneH + safezoneY;
        w = 0.098 * safezoneW;
        h = 0.025 * safezoneH;
      };
    };
};


MissionName.MapName/scarCODE/scarHUD/hpp/RscScarHudCurrentWeapon.hpp
class RscScarHudCurrentWeapon
{
    idd = 16032;
    duration = 99999;
    fadeIn = 0.3;
    onLoad = "uiNamespace setVariable ['RscScarHudCurrentWeapon', _this select 0]";
    class IGUIBack
    {
    	type = 0;
    	style = 128;
    	text = "";
    	colorText[] = {0,0,0,0};
    	font = "PuristaMedium";
    	sizeEx = 0;
    	shadow = 0;
    	colorbackground[] = {1,1,1,0.97};
    };
    class RscText
    {
    	deletable = 0;
    	fade = 0;
    	access = 0;
    	type = 0;
    	colorBackground[] = {1,1,1,0.97};
    	colorText[] = {0,0,0,0.8};
    	text = "";
    	fixedWidth = 0;
    	style = 0;
    	shadow = 0;
    	colorShadow[] = {0,0,0,0.5};
    	font = "PuristaMedium";
    	SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1)";
    	linespacing = 1;
    	tooltipColorText[] = {1,1,1,1};
    	tooltipColorBox[] = {1,1,1,1};
    	tooltipColorShade[] = {0,0,0,0.65};
    };
    class RscPicture
    {
        deletable = 0;
    	fade = 0;
    	access = 0;
    	type = 0;
    	style = 48;
    	colorBackground[] = {0,0,0,0};
    	colorText[] = {0,0,0,0.8};
    	font = "TahomaB";
    	sizeEx = 0;
    	lineSpacing = 0;
    	text = "";
    	fixedWidth = 0;
    	shadow = 0;
    	tooltipColorText[] = {1,1,1,1};
    	tooltipColorBox[] = {1,1,1,1};
    	tooltipColorShade[] = {0,0,0,0.65};
    };
    class Controls
    {
      class stanceBackground: IGUIBack
      {
        idc = 1007;
        x = 0.956 * safezoneW + safezoneX;
        y = 0.884 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.0685 * safezoneH;
      };
      class stanceImage: RscPicture
      {
        idc = 1002;
        style = 2096;
        x = 0.956 * safezoneW + safezoneX;
        y = 0.888 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.045 * safezoneH;
      };
      class playerDamage: RscText
      {
        idc = 1006;
        style = 2;
        font = "PuristaSemiBold";
        colorBackground[] = {1,1,1,0.3};
        x = 0.956 * safezoneW + safezoneX;
        y = 0.935 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.018 * safezoneH;
      };
      class bulletCountBackground: IGUIBack
      {
        idc = -1;
        x = 0.889 * safezoneW + safezoneX;
        y = 0.917 * safezoneH + safezoneY;
        w = 0.065 * safezoneW;
        h = 0.036 * safezoneH;
      };
      class bulletCount: RscText
      {
        idc = 1001;
        style = 1;
        font = "PuristaSemiBold";
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 2.7)";
        colorBackground[] = {-1,-1,-1,-1};
        x = 0.889 * safezoneW + safezoneX;
        y = 0.911 * safezoneH + safezoneY;
        w = 0.065 * safezoneW;
        h = 0.04 * safezoneH;
      };
      class currentZeroing: RscText
      {
        idc = 1003;
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 0.8)";
        x = 0.856 * safezoneW + safezoneX;
        y = 0.915 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.018 * safezoneH;
      };
      class currentWeaponMode: RscText
      {
        idc = 1004;
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 0.8)";
        x = 0.856 * safezoneW + safezoneX;
        y = 0.935 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.018 * safezoneH;
      };
      class magazineCount: RscText
      {
        idc = 1005;
        style = 1;
        font = "PuristaBold";
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1.6)";
        x = 0.889 * safezoneW + safezoneX;
        y = 0.892 * safezoneH + safezoneY;
        w = 0.065 * safezoneW;
        h = 0.025 * safezoneH;
      };
      class currentThrowable: RscText
      {
        x = 0.856 * safezoneW + safezoneX;
        y = 0.956 * safezoneH + safezoneY;
        w = 0.1315 * safezoneW;
        h = 0.02 * safezoneH;

        idc = 1010;
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 0.7)";
        style = 1;
      };
    };
};


MissionName.MapName/scarCODE/scarHUD/hpp/RscScarHudCurrentVehicle.hpp
class RscScarHudCurrentVehicle
{
    idd = 16031;
    duration = 99999;
    fadeIn = 0.3;
    onLoad = "uiNamespace setVariable ['RscScarHudCurrentVehicle', _this select 0]";
    class IGUIBack
    {
    	type = 0;
    	style = 128;
    	text = "";
    	colorText[] = {0,0,0,0};
    	font = "PuristaMedium";
    	sizeEx = 0;
    	shadow = 0;
    	colorbackground[] = {1,1,1,0.97};
    };
    class RscText
    {
    	deletable = 0;
    	fade = 0;
    	access = 0;
    	type = 0;
        colorBackground[] = {1,1,1,0.97};
    	colorText[] = {0,0,0,0.8};
    	text = "";
    	fixedWidth = 0;
    	style = 0;
    	shadow = 0;
    	colorShadow[] = {0,0,0,0.5};
    	font = "PuristaMedium";
    	SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1)";
    	linespacing = 0;
    	tooltipColorText[] = {1,1,1,1};
    	tooltipColorBox[] = {1,1,1,1};
    	tooltipColorShade[] = {0,0,0,0.65};
    };
    class RscPicture
    {
        deletable = 0;
    	fade = 0;
    	access = 0;
    	type = 0;
    	style = 48;
    	colorBackground[] = {0,0,0,0};
    	colorText[] = {0,0,0,0.8};
    	font = "TahomaB";
    	sizeEx = 0;
    	lineSpacing = 0;
    	text = "";
    	fixedWidth = 0;
    	shadow = 0;
    	tooltipColorText[] = {1,1,1,1};
    	tooltipColorBox[] = {1,1,1,1};
    	tooltipColorShade[] = {0,0,0,0.65};
    };
    class Controls
    {
      class vehInfoPanel: IGUIBack
      {
        idc = -1;
        x = 0.872 * safezoneW + safezoneX;
        y = 0.931 * safezoneH + safezoneY;
        w = 0.015 * safezoneW;
        h = 0.045 * safezoneH;
      };
      class vehSpeedBackground: IGUIBack
      {
        idc = -1;
        x = 0.889 * safezoneW + safezoneX;
        y = 0.917 * safezoneH + safezoneY;
        w = 0.065 * safezoneW;
        h = 0.036 * safezoneH;
      };
      class vehicleSpeed: RscText
      {
        idc = 1000;
        style = 1;
        font = "PuristaSemiBold";
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 2.7)";
        colorBackground[] = {-1,-1,-1,-1};
           x = 0.889 * safezoneW + safezoneX;
        y = 0.911 * safezoneH + safezoneY;
        w = 0.065 * safezoneW;
        h = 0.04 * safezoneH;
      };
      class vehicleImageBackground: IGUIBack
      {
        idc = 1001;
        x = 0.956 * safezoneW + safezoneX;
        y = 0.884 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.0685 * safezoneH;
      };
      class vehicleImage: RscPicture
      {
        idc = 1002;
        style = 2096;
        x = 0.956 * safezoneW + safezoneX;
        y = 0.888 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.045 * safezoneH;
      };
      class vehicleDamage: RscText
      {
        idc = 1003;
        style = 2;
        font = "PuristaSemiBold";
        colorBackground[] = {1,1,1,0.3};
        x = 0.956 * safezoneW + safezoneX;
        y = 0.935 * safezoneH + safezoneY;
        w = 0.031 * safezoneW;
        h = 0.018 * safezoneH;
      };
      class ALT: RscText
      {
        idc = 1006;
        style = 1;
        fade = 1;
        font = "PuristaSemiBold";
        SizeEx = "(((((safezoneW / safezoneH) min 1.2) / 1.2) / 25) * 1.6)";
        x = 0.889 * safezoneW + safezoneX;
        y = 0.892 * safezoneH + safezoneY;
        w = 0.065 * safezoneW;
        h = 0.025 * safezoneH;
      };
      class vehicleFuelBarBackground: IGUIBack
      {
        idc = -1;
        x = 0.915 * safezoneW + safezoneX;
        y = 0.956 * safezoneH + safezoneY;
        w = 0.0725 * safezoneW;
        h = 0.02 * safezoneH;
      };
      class vehicleFuelBar: RscText
      {
        idc = 1010;
        type = 8;
        colorBar[] = {0,0,0,0.8};
        colorFrame[] = {-1,-1,-1,-1};
        texture = "#(argb,8,8,3)color(1,1,1,1)";
        x = 0.9165 * safezoneW + safezoneX;
        y = 0.958 * safezoneH + safezoneY;
        w = 0.0695 * safezoneW;
        h = 0.015 * safezoneH;
      };
      class txtFuel: RscText
      {
        idc = -1;
        style = 2;
        text = "FUEL";
        x = 0.889 * safezoneW + safezoneX;
        y = 0.956 * safezoneH + safezoneY;
        w = 0.025 * safezoneW;
        h = 0.02 * safezoneH;
      };
      class imgLights: RscPicture
      {
        idc = 1011;
        style = 2096;
        text = "A3\ui_f\data\IGUI\Cfg\VehicleToggles\lightsiconon_ca.paa";
        colorText[] = {0,0,0,0.25};
        x = 0.872 * safezoneW + safezoneX;
        y = 0.932 * safezoneH + safezoneY;
        w = 0.015 * safezoneW;
        h = 0.025 * safezoneH;
      };
      class imgCollisLights: RscPicture
      {
        idc = 1012;
        style = 2096;
        text = "A3\ui_f\data\IGUI\Cfg\VehicleToggles\collisionlightsiconon_ca.paa";
        colorText[] = {0,0,0,0.25};
        x = 0.872 * safezoneW + safezoneX;
        y = 0.95 * safezoneH + safezoneY;
        w = 0.015 * safezoneW;
        h = 0.025 * safezoneH;
      };
    };
};
